cloud-sql/r2dbc/README.md

```diff
@@ -0,0 +1,104 @@
+# Connecting to Cloud SQL - MySQL and Postgres
+
+## Before you begin
+
+1. If you haven't already, set up a Java Development Environment (including google-cloud-sdk and
+maven utilities) by following the [Java setup guide](https://cloud.google.com/java/docs/setup) and
+[creating a project](https://cloud.google.com/resource-manager/docs/creating-managing-projects#creating_a_project).
+
+1. You can use MySQL or PostgreSQL instance for this sample. 
+Create a 2nd Gen Cloud SQL Instance by following corresponding instructions: 
+[MySQL](https://cloud.google.com/sql/docs/mysql/create-instance) /
+[PostgreSQL](https://cloud.google.com/sql/docs/postgres/create-instance).
+Note the connection string, database user, and database password that you create. 
+
+1. Create a database for your application by following corresponding instructions:
+[MySQL](https://cloud.google.com/sql/docs/mysql/create-manage-databases) /
+[PostgreSQL](https://cloud.google.com/sql/docs/postgres/create-manage-databases).
+Note the database name.
+
+1. Assign your connection details in the following format:
+
+    ```
+    r2dbc:pool:gcp:<'mysql' or 'postgres'>://<user>:<password>@<connection_name>/<db_name>[?connectionOption1=optionValue1[&connectionValue2=optionValue2]]
+    ```
+    to an environment variable `CLOUD_SQL_CONNECTION_STRING`.
+
+    Example for MySQL:
+    ```sh
+    export CLOUD_SQL_CONNECTION_STRING="r2dbc:pool:gcp:mysql://user:123456@my-project:us-central1:r2dbctest/testdb?maxIdleTime=PT60S&initialSize=2&maxSize=12" 
```

Comment:
I'd prefer if the connection options were set programatically within the sample itself rather than in the readme, so that we can include them in the Cloud Docs. Something like:
```
ConnectionPoolConfiguration configuration = ConnectionPoolConfiguration.builder(connectionFactory)
   .maxIdleTime(Duration.ofMillis(1000))
   .maxSize(20)
   .build();

ConnectionPool pool = new ConnectionPool(configuration);
```